src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -73,67 +73,60 @@ where TYPE(ex) = QuizExercise
     /**
      * Query which fetches all the exams for which the user is instructor in the course and matching the search criteria.
      *
-     * @param partialTitle     exam title search term
-     * @param partialExamTitle course title search term
-     * @param groups           user groups
-     * @param pageable         Pageable
+     * @param searchTerm search term
+     * @param groups     user groups
+     * @param pageable   Pageable
      * @return Page with search results
      */
     @Query("""
             SELECT DISTINCT e FROM Exam e
                     WHERE e.course.instructorGroupName IN :groups
-                    AND (e.title LIKE %:partialTitle% OR e.course.title LIKE %:partialExamTitle%)
+                    AND (CONCAT(e.id, '') = :#{#searchTerm} OR e.title LIKE %:partialTitle% OR e.course.title LIKE %:partialExamTitle%)
```

Comment:
Afaik this should work as well:
```suggestion
                    AND (LTRIM(e.id) = :#{#searchTerm} OR e.title LIKE %:partialTitle% OR e.course.title LIKE %:partialExamTitle%)
```
Same for the other occurrences.

Another note on this:
And while it might not be that a bit of concern here, converting the id to a string for comparison is rather inefficient. At least in Postgres this leads to a sequence scan instead of an index scan. This, in turn, leads to way worse performance of the query. So consider testing the query performance and see if you can maybe instead try to convert the string to a long instead (if possible).

Reply:
Are you sure that is a problem here? Because of the other parts of the query no index usage is possible any way?

Reply:
Edited my comment.
The LTRIM(...) is simply shorter than CONCAT(..., ''), which is why I suggested it.

Reply:
Changed it to ltrim in 433b9651343162b17f2ce8f8e3b5a8a323b27e30